watchdog/main.go

```diff
@@ -232,6 +232,34 @@ func getAdditionalEnvs(config *WatchdogConfig, r *http.Request, method string) [
 	return envs
 }
 
+func lockFilePresent() bool {
+	path := filepath.Join(os.TempDir(), ".lock")
```

Comment:
@alexellis this jumped out at me and having a look at the rest of the Watchdog code it doesn't appear to check/ensure that the temp directory actually exists before attempting to write a lock..

If the user attempts to use the watchdog within a Scratch container then it'll bomb out.
```
2018/03/17 18:03:15 Writing lock-file to: /tmp/.lock
2018/03/17 18:03:15 Cannot write /tmp/.lock. To disable lock-file set env suppress_lock=true.
```
Or am I missing something here?

Reply:
@johnmccabe @alexellis does this behaviour needs to be changed?

Reply:
It will crash. Having a temporary directory is a requirement.